static/app/components/timezoneProvider.tsx

```diff
@@ -0,0 +1,90 @@
+import {createContext, useContext, useEffect, useMemo, useState} from 'react';
+
+import {useUser} from 'sentry/utils/useUser';
+
+interface TimezoneProviderValue {
+  timezone: string;
+  setOverride?: (timezone: string | null) => void;
```

Comment:
why optional? just make two types so callsites don't need to check if it exists

Reply:
what would that look like?